stdlib/public/core/Range.swift

```diff
@@ -14,7 +14,7 @@
 
 public struct RangeOfStrideable<
   Element : Strideable
-> : Equatable, Collection,
+> : Equatable, RandomAccessCollection,
```

Comment:
> The following exists against Collection however they are commented out currently because of the noted compiler issues.

You're right -- the default implementation will take care of this!